pkg/analyzer/analyzer.go

```diff
@@ -390,10 +391,38 @@ func (a *analyzerInfo) worker(results, unwanted chan<- string, locCount chan<- i
 			results <- ansible
 			locCount <- linesCount
 		}
-	// It could be Ansible, Buildah, CloudFormation, Crossplane, or OpenAPI
+	// It could be Ansible, Buildah, CloudFormation, Crossplane, OpenAPI or CICD
 	case yaml, yml, json, sh:
-		a.checkContent(results, unwanted, locCount, linesCount, ext)
+		if isGithubWorkflowfile(a.filePath) {
```

Comment:
since this function will read the file and check regexes for the content, the call to it should be inside checkContent function. as the file is already read there.